client-js/app/elements/labrad-registry.ts

```diff
@@ -42,15 +43,76 @@ export class LabradRegistry extends polymer.Base {
                                               this.$.editValueInput);
   }
 
+
+  getDefaultSelectedItem(): number {
+    return this.path.length > 0 ? 1 : 0;
+  }
+
+
+  cursorMove(e) {
+    const offset = this.path.length > 0 ? 1 : 0;
+    const length = this.keys.length + this.dirs.length + offset;
+
+    switch (e.detail.combo) {
+      case 'up':
+        if (this.selectedIdx === null || this.selectedIdx === 0) {
+          this.set('selectedIdx', null);
+        } else {
+          this.set('selectedIdx', this.selectedIdx - 1);
+        }
+        break;
```

Comment:
Can you clarify the issue? The later part of your statement makes it just sound like if you scroll off where you've selected, then use the keys, nothing happens because you are adjusting the selection off-screen. The earlier part makes it sound like the selection gets out of sync or something though, so it's unclear what you are saying.